students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+                    break
+                else:
+                    print("Integer selected is not a valid option. "
+                              "Please, enter max or min. \n")
+        return str_user_input
+
+########### END OF MENU CLASS###############
+
+# Methods defined in the files class are related to opening and renaming files.
+# Modifications to the methods are needed if the files are not for a task priority list.
+
+
+class Files:
+    @staticmethod
+    def write_email(donors):
+        '''
+        Writes emails to the donors and saves them to text files.
+        :param donors: Dictionary that stores all the donor's donation information.
+        :return:
+        '''
+        for donor_name in donors:
+            # Open text file and assign a mode to the file.
+            new_file = open(donor_name + ".txt", mode="w")
+            new_file.write(Files.donor_email(donor_name))
+            new_file.close()
+            print(donor_name + " saved to "+donor_name+".txt.")
+
+    @staticmethod
+    def donor_email(donor):
+        '''
+        Formats the email with donor's name and total lifetime donations.
+        :param donor: Dictionary that stores all the donor's donation information.
+        :param dictionary: A dictionary that stores the donor as the 'key' and the lifetime donation as the 'value'.
+        :return: The concatenation how the three strings.
+        '''
+        str_salutation = "Dear {},".format(donor)
+        str_body = "\n\n"+"Thank you for generosity! " \
+                   "\n\n"+"This donation will be put to great use!" \
+                   "\n\n"+"Your current lifetime donation is "+"${:,.02f}".format(dic_donors[donor][0]) + "."
+        str_valediction = "\n\n"+"Sincerely," \
+                          "\n\n"+"The Team"
+        thankyou_message = str_salutation+str_body+str_valediction
+        return thankyou_message
+
+############# END OF FILE CLASS #################
+
+# The Data class is where the data get manipulated.
+
+
+class Data:
+    @staticmethod
+    def update_donor(donor_name, donation_amount, initial_donation):
+        '''
+        Updates donor's donation history or creates a new donor.
+        :param donor_name: The name of the donor.
+        :param donation_amount: The amount of money the donor donated.
+        :param initial_donation: The initial donation of the donor.
+        :return:
+        '''
+        # Creates a donor name list
+        # lst_donor_name = [dic_donors[row][0] for row in range(len(dic_donors))]  # --> Added list comprehension
+        # Checks to see if donor has given before and updates history
+        dic_new_donor_names = dic_donors
+        if donor_name in dic_donors:
+            dic_new_donor_names[donor_name][1] = dic_donors[donor_name][
+                                                     1] + 1  # --> Increase amount of times donor donated
```

Comment:
You don't need to keep track of the total and initial donations in the list. If you store only the individual donations in a list, you can use `sum()` for total and keep the donations sorted in the order they were donated to access the initial donation by `0` index. 

If you still want to keep track of these calculated values elsewhere on a per donor basis (which is a great idea), it would be best to use `@properties` on your Donor objects